src/Homework/Week3/Pb2/Customer.java

```diff
@@ -0,0 +1,58 @@
+package Homework.Week3.Pb2;
+
+import java.util.List;
+
+public class Customer {
+    private String name;
+    private MemberShip myMemberShip;
+    {
+        myMemberShip = MemberShip.NONE;
+    }
+    private Basket myBasket;
+
+    Customer(String name, MemberShip memberShip)
+    {
+        myMemberShip = memberShip;
+        myBasket = new Basket();
+    }
+
+    void addToBasket(ItemForSale item) {
+        if(item == null) {
+            System.out.println("The item added to the basket is null");
+            return;
+        }
+        myBasket.addItem(item);
+    }
+
+    float basketPrice()
+    {
+        if(myMemberShip.equals(MemberShip.NONE))
+            return myBasket.getTotalPrice();
+        else if(myMemberShip.equals(MemberShip.SILVER))
+            return myBasket.getTotalPrice()*90/100;
```

Comment:
o alta recomadare pentru a fi mai flexibil de implementat codul pe viitor: poti tine discountul direct la nivelul enum-ului(un field private discount si un getter) si astfel ai avea un loc centralizat unde e implementata aceasta relatie dintre un membership si discountul asociat, de care te poti folosi apoi in mai multe locuri din cod; daca pe viitor o sa fie nevoie sa faci o schimbare(sa schimbi discountul de la 20% la 25% pt membershipul GOLD), vei face schimbarea doar acolo si se va propaga pe oriunde prin cod